src/doc/layer.cpp

```diff
@@ -36,6 +36,9 @@ Layer::Layer(ObjectType type, Sprite* sprite)
          type == ObjectType::LayerTilemap);
 
   setName("Layer");
+  // Always generate a UUID for this layer, but take into account that it could
+  // be replaced. For instance, when loading a layer that already had a UUID.
+  m_uuid = base::Uuid::Generate();
```

Comment:
Probably I'd prefer to generate the `m_uuid` in the `uuid()` getter when `m_uuid` is empty (`m_uuid == Uuid()`).
```c++
const base::Uuid& Layer::uuid() const
{
  if (m_uuid == base::Uuid())
    m_uuid = base::Uuid::Generate();
  return m_uuid;
}
```
Converting `m_uuid` in `mutable`.